src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -611,10 +598,34 @@ BoundExpression convertArrayElement(BoundExpression element, TypeSymbol elementT
             // PROTOTYPE: Improve perf. For instance, emit [] as Array.Empty<T>() rather than a List<T>.
             var result = BindCollectionInitializerCollectionLiteral(
                 node,
+                CollectionLiteralTypeKind.ListInterface,
                 GetWellKnownType(WellKnownType.System_Collections_Generic_List_T, diagnostics, node.Syntax).Construct(elementType),
                 wasCompilerGenerated: wasCompilerGenerated,
                 diagnostics);
-            return result.Update(result.Placeholder, result.CollectionCreation, result.Elements, targetType);
+            return result.Update(result.CollectionTypeKind, result.Placeholder, result.CollectionCreation, result.Elements, targetType);
+        }
+
+        private BoundCollectionLiteralExpression BindCollectionLiteralForErrorRecovery(
+            BoundUnconvertedCollectionLiteralExpression node,
+            TypeSymbol targetType,
+            BindingDiagnosticBag diagnostics)
+        {
+            var syntax = node.Syntax;
+            var builder = ArrayBuilder<BoundExpression>.GetInstance(node.Elements.Length);
+            foreach (var element in node.Elements)
+            {
+                var result = BindToNaturalType(element, diagnostics);
+                result.WasCompilerGenerated = true;
```

Comment:
is this needed for IOperation?

Reply:
No, we shouldn't be setting `WasCompilerGenerated` here. Removed.

Reply:
Got it. I do want to make sure that disabling the IOperation tests isn't masking an actual need for this. But in either case we can address when we get around to fixing IOperation tests.